### PR TITLE
Prevent unwrapped names via re-registering previously wrapped names

### PR DESCRIPTION
**Pull request by jefflau**
* Fixes issue where registering a previously wrapped name allows ownership on it in the NameWrapper

### CODE REVIEW

File: contracts/wrapper/NameWrapper.sol [hunk 1]
```diff
@@ -129,7 +129,19 @@ contract NameWrapper is
 
         bytes32 labelhash = _getEthLabelhash(bytes32(id), fuses);
         if (labelhash != bytes32(0)) {
-            expiry = uint64(registrar.nameExpires(uint256(labelhash))) + GRACE_PERIOD;
+            expiry =
+                uint64(registrar.nameExpires(uint256(labelhash))) +
+                GRACE_PERIOD;
+            // if owner in registrar is not the wrapper, zero out the owner
+            try registrar.ownerOf(uint256(labelhash)) returns (
```
Comment (Arachnid) [inline contracts/wrapper/NameWrapper.sol:136]: You can probably avoid the need for the try/catch if you just check if expiry + grace period is in the future.

File: contracts/wrapper/NameWrapper.sol [hunk 2]
```diff
@@ -127,9 +127,19 @@ contract NameWrapper is
...
-        if (labelhash != bytes32(0)) {
-            expiry = uint64(registrar.nameExpires(uint256(labelhash))) + GRACE_PERIOD;
+        bytes32 labelHash = _getEthLabelhash(bytes32(id), fuses);
+        if (labelHash != bytes32(0)) {
+            uint64 registrarExpiry = uint64(
+                registrar.nameExpires(uint256(labelHash))
+            );
+            expiry = registrarExpiry + GRACE_PERIOD;
+            // if owner in registrar is not the wrapper, zero out the owner
+            if (
```
Comment (Arachnid) [inline contracts/wrapper/NameWrapper.sol:137]: Hm, does this mean that a name that's subject to the vulnerability will briefly appear to have an owner again during the grace period? We should probably zero out the owner if the expiry is in the past even if it *is* in grace.

Comment (jefflau) [reply contracts/wrapper/NameWrapper.sol:137]: Isn't this already doing that?

Comment (Arachnid) [reply contracts/wrapper/NameWrapper.sol:137]: My mistake.

**Review by Arachnid — APPROVED** [review]

**Pull request merged**